### PR TITLE
Donot reserve string column size and remove uncertain size reservation

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: ref #6233 

Problem Summary:

### What is changed and how it works?
For Join Probe, we can't guess the output column size, thus remove the reserve operations.
For V0 data packet, string column will reserve 64 bytes per element, often too large. Don't reserve memory for it.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp [hunk 1]
```diff
@@ -170,10 +171,16 @@ Block CHBlockChunkCodec::decodeImpl(ReadBuffer & istr, size_t reserve_size)
...
         /// Data
         MutableColumnPtr read_column = column.type->createColumn();
-        if (reserve_size > 0)
-            read_column->reserve(std::max(rows, reserve_size));
-        else if (rows)
-            read_column->reserve(rows);
+        /// For non-fixed size type, reserve function might cause too much memory usage, i.e. string type column reserves 64 bytes
+        /// size for each element.
+        const auto & type_removed_nullable = removeNullable(column.type);
+        if (type_removed_nullable->isValueRepresentedByNumber() || type_removed_nullable->isFixedString())
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp:177]: Maybe add a function `reservable`, and add some comments why some data types cannot be reserved.

**Review by SeaRise — COMMENTED** [review]
Others LGTM

File: dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp (hunk 1 above)
Comment (yibin87) [reply dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp:177]: Make sense, I'll add it

Comment (yibin87) [reply dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp]: Found one existing interface "haveMaximumSizeOfValue" exactly match the request. Use it instead.

**Review by SeaRise — APPROVED** [review]
Others LGTM

File: dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp [hunk 2]
```diff
@@ -170,10 +171,14 @@ Block CHBlockChunkCodec::decodeImpl(ReadBuffer & istr, size_t reserve_size)
 
         /// Data
         MutableColumnPtr read_column = column.type->createColumn();
-        if (reserve_size > 0)
-            read_column->reserve(std::max(rows, reserve_size));
-        else if (rows)
-            read_column->reserve(rows);
+        const auto & type_removed_nullable = removeNullable(column.type);
+        if (type_removed_nullable->haveMaximumSizeOfValue())
```
[the remark this answers is not in the document]
Comment (yibin87) [reply dbms/src/Flash/Coprocessor/CHBlockChunkCodec.cpp]: Good catch, update it.

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**